File: CPP/src/engine/core/Time.cpp
//
// Created by Razim on 1/6/2016.
//

#include <chrono>
#include "Time.h"

long Time::second = 1000;
double Time::delta;

void Time::setDelta(double delta) {
    Time::delta = delta;
}

double Time::getDelta(){
    return Time::delta;
}

long Time::getTime() {
    unsigned long milliseconds_since_epoch = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
    return milliseconds_since_epoch;
}

File: JAVA/src/main/scala/engine/core/Time.scala
package engine.core

/**
  * Created by Mnenmenth Alkaborin
  * Please refer to LICENSE file if included
  * for licensing information
  * https://github.com/Mnenmenth
  */
object Time {

  val second: Long = 1000000000

  var delta: Double = 0

  def getTime = System.nanoTime()

  def setDelta(delta: Double) = Time.delta = delta

}
